### DISCUSSION TITLE
Consideration for implementing subscriptions via the Open Source 'Lago' project to reduce Pay Vendor Lockin.

### DISCUSSION THREAD

**Discussion by evolve2k** [pay-rails/pay#808] [General]
I'm bringing this up as while we've used the pay gem, we've essentially become vendor locked to stripe by having our subscription management stuck in there. Making it difficult to have subscriptions linked to other payment providers.

Lago is a new "Open-source metering and usage-based billing" project. 
ref: https://news.ycombinator.com/item?id=31424450

Lago can be [connected to any Payment Service Provider (PSP)](https://doc.getlago.com/docs/guide/payments/overview) but then abstracts advanced subscription and meter-biling management all into an open source self hosted platform. 

For stripe users, the shift would mean keeping Stripe Payments to charge credit cards, but bypassing Stribe Billing that charges an extra fee to manage subscriptions.

Subscriptions could be internally managed with your project, regardless of the payment source used and it would be much easier to then add more payment sources with different Payment Service Providers (such as those in specific countries not supported by Stripe/Braintree etc).

Currently the pay gem is positioned as an abstraction over payments and subscriptions, this would allow further uncoupling from the payment provider by supporting subscription management through a self hosted lato instance.

Interested in others thoughts.

I found this page provides the best clarity on the difference from just implementing Stripe Billing Subscriptions directly.
https://www.getlago.com/resources/compare/compare-home

Ps. Im not involved in the project, but keep coming back to it each time I have vendor issues with Stripe.

**Comment by excid3** [comment]
I've seen Lago a few times and haven't tried it yet. If someone wants to start on an implementation, that would be awesome.

**Comment by evolve2k** [comment]
_Thanks Chris._

I've reached out to the Lago team and added it as a feature request on their end (unless someone else gets to it first).
Folks you can vote for support for this from the lago team here: 

https://getlago.canny.io/feature-requests/p/integration-with-ruby-pay-gem
